core/src/test/java/io/questdb/griffin/SqlParserTest.java

```diff
@@ -5044,6 +5044,21 @@ public void testNoopGroupByFailureWhenUsing2KeysInSelectStatementButOnlyOneInGro
         );
     }
 
+    @Test
+    public void testNoopGroupByFailureWhenUsing2KeysInSelectStatementButOnlyOneInGroupByV2() throws Exception {
+        assertFailure(
+                "select sym1, sym2, avg(bid) avgBid from x where sym1 in ('AA', 'BB' ) group by sym2",
```

Comment:
may be tests pass all the time then? :)

i just modified this test:

```java
    @Test
    public void testNoopGroupByWith2Syms() throws Exception {
        assertQuery(
                "select-group-by sym1, sym2, avg(bid) avgBid from (select [sym1, sym2, bid] from x timestamp (ts) where sym1 in ('AA','BB'))",
                "select sym1, sym2, avg(bid) avgBid from x where sym1 in ('AA', 'BB' ) group by sym1",
                modelOf("x")
                        .col("sym1", ColumnType.SYMBOL)
                        .col("sym2", ColumnType.SYMBOL)
                        .col("bid", ColumnType.INT)
                        .col("ask", ColumnType.INT)
                        .timestamp("ts")
        );
    }

```

and it also passes. Why is that?

Reply:
ok, issue here is that SqlParserTest.assertQuery() does not call SqlCodeGenerator , which is where the exception is thrown. I'll fix the tests now